pkg/tsdb/cloudmonitoring/time_series_filter.go

```diff
@@ -220,13 +223,24 @@ func (timeSeriesFilter *cloudMonitoringTimeSeriesFilter) parseResponse(queryRes
 							timeField,
 							valueField,
 						},
+						Meta: &data.FrameMeta{
+							ExecutedQueryString: executedQueryString,
```

Comment:
Same here. Would you add a test or add to an existing test the use of this field ExecutedQueryString?

Reply:
I add assert for this field.

Reply:
This field is not directly asserted in these tests (or any other tests here :( ). 
This is probably out of scope for this PR, but what kind of data enters this condition on line 208? It would be great to have a test for it, but that is not blocking for this PR.

Reply:
Ah, sorry. I only added test which using test data 3, but I should add test which using test data 4 too.

Reply:
Thanks!

Reply:
Hmm, [your test ](https://github.com/grafana/grafana/pull/45500/files#diff-c8d1bd71f56615fc3c3cc3afb8b40b9c1da7805fdf3c97527ae11d8e66f9a4d9R417) with test data 4 doesn't seem to enter that condition still.

Reply:
I misunderstand something, I didn't read the code carefully...
I'll try to understand the condition...

Reply:
The code is added by me...
https://github.com/grafana/grafana/commit/6770f2e9401547863048f58d88a52e3e9dbcb975#diff-19faf88f206b68e7fad5a3f06ee85a43d6d67abe25edcd38343396a09157e431R356-R394
I completely forgot it.

And, now I remember the condition.

When `parseFloat()` is failed at ![here](https://github.com/grafana/grafana/blob/v8.4.1/pkg/tsdb/cloudmonitoring/time_series_filter.go#L173), the ![next for clause](https://github.com/grafana/grafana/blob/v8.4.1/pkg/tsdb/cloudmonitoring/time_series_filter.go#L204-L225
) will be executed.

Sample test data.
 https://gist.github.com/mtanda/e7d33c2d7e742b6e4b0fba2239b52608#file-gistfile1-txt-L101

But I didn't remember that why I ignore `parseFloat()` error.
I'm not sure Cloud Monitoring actually return such broken response.

Reply:
According to ![doc](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TypedValue#Distribution), the `bucketCounts` is int64 format string.
I think it is safe the for clause, but it might be out of scope of this PR.

Reply:
Really sorry, there is a misunderstanding. I'm asking about the next `for` loop, not related to the `ParseFloat` error. Right [here](https://github.com/grafana/grafana/pull/45500/files#diff-be823fa860127465e4c9518eab333dec6bbac9b158c01a59f62ac302cd825269R207) at line 207 of this PR.

Reply:
Yes, I told about the line 227. The code is not executed if `ParseFloat` doesn't return error.
Sorry, my explanation is might be bad...

Reply:
Ok I see. No problem, I might have misunderstood. 
Here's what I believe I see:
* `ParseFloat` returns error
* `continue` sends it back to [line 170](https://github.com/grafana/grafana/pull/45500/files#diff-be823fa860127465e4c9518eab333dec6bbac9b158c01a59f62ac302cd825269R170)
* when line 170's `for` loop finishes, the outer `for` [line 164](https://github.com/grafana/grafana/pull/45500/files#diff-be823fa860127465e4c9518eab333dec6bbac9b158c01a59f62ac302cd825269R164) continues to `for` on [line 207](https://github.com/grafana/grafana/pull/45500/files#diff-be823fa860127465e4c9518eab333dec6bbac9b158c01a59f62ac302cd825269R207)
* I don't know how the `if` condition on [line 208](https://github.com/grafana/grafana/pull/45500/files#diff-be823fa860127465e4c9518eab333dec6bbac9b158c01a59f62ac302cd825269R208) is entered

So what I understand is that the line 207 `for` loop is entered no matter if there is an error returned or not from `ParseFloat`, but not sure what kind of data enters the `if` on line 208.

Reply:
When I implemented this code, I might hit the data, but I don't remember it...
I'll check Cloud Monitoring API response in my environment.

Reply:
This prometheus exporter expect bucketCounts is integer.
https://github.com/prometheus-community/stackdriver_exporter/blob/v0.12.0/collectors/monitoring_collector.go#L501


Old Cloud Monitoring document says, bucketCounts is int64 format.
https://web.archive.org/web/20170226132705/https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TypedValue
> string (int64 format)
> If bucketOptions is given, then the sum of the values in bucketCounts must equal the value in count. If bucketOptions is not given, no bucketCounts fields may be given.
> Bucket counts are given in order under the numbering scheme described above (the underflow bucket has number 0; the finite buckets, if any, have numbers 1 through N-2; the overflow bucket has number N-1).
> The size of bucketCounts must be no greater than N as defined in bucketOptions.
> Any suffix of trailing zero bucketCount fields may be omitted.

I might have misunderstood, the code seems to be safe to remove.